runtime/mainnet/src/config/contracts.rs

```diff
@@ -0,0 +1,227 @@
+use frame_support::{
+	parameter_types,
+	traits::{ConstBool, ConstU32, ConstU64, Nothing},
+};
+use frame_system::EnsureSigned;
+
+use crate::{
+	deposit, Balance, Balances, Perbill, Runtime, RuntimeCall, RuntimeEvent, RuntimeHoldReason,
+	Timestamp, UNIT,
+};
+
+const ETH: u128 = 1_000_000_000_000_000_000;
+
+parameter_types! {
+	// TODO: review implications of this
+	pub const DepositPerItem: Balance = deposit(1, 0);
+	// TODO: review implications of this
+	pub const DepositPerByte: Balance = deposit(0, 1);
+	pub CodeHashLockupDepositPercent: Perbill = Perbill::from_percent(30);
+	pub const NativeToEthRatio: u32 = (ETH/UNIT) as u32;
+}
+
+impl pallet_revive::Config for Runtime {
+	type AddressMapper = pallet_revive::AccountId32Mapper<Self>;
+	type CallFilter = Nothing;
+	type ChainExtension = ();
+	type ChainId = ConstU64<3395>;
+	type CodeHashLockupDepositPercent = CodeHashLockupDepositPercent;
+	type Currency = Balances;
+	type Debug = ();
+	type DepositPerByte = DepositPerByte;
+	type DepositPerItem = DepositPerItem;
+	type InstantiateOrigin = EnsureSigned<Self::AccountId>;
+	type NativeToEthRatio = NativeToEthRatio;
+	// TODO: review implications of this
+	type PVFMemory = ConstU32<{ 512 * 1024 * 1024 }>;
+	type RuntimeCall = RuntimeCall;
+	type RuntimeEvent = RuntimeEvent;
+	type RuntimeHoldReason = RuntimeHoldReason;
+	// TODO: review implications of this
+	type RuntimeMemory = ConstU32<{ 128 * 1024 * 1024 }>;
+	type Time = Timestamp;
+	type UnsafeUnstableInterface = ConstBool<false>;
+	type UploadOrigin = EnsureSigned<Self::AccountId>;
+	type WeightInfo = pallet_revive::weights::SubstrateWeight<Self>;
+	type WeightPrice = pallet_transaction_payment::Pallet<Self>;
+	type Xcm = pallet_xcm::Pallet<Self>;
+}
+
+impl TryFrom<RuntimeCall> for pallet_revive::Call<Runtime> {
+	type Error = ();
+
+	fn try_from(value: RuntimeCall) -> Result<Self, Self::Error> {
+		match value {
+			RuntimeCall::Contracts(call) => Ok(call),
+			_ => Err(()),
+		}
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	use std::any::TypeId;
+
+	use frame_support::pallet_prelude::Get;
+
+	use super::*;
+
+	// 18 decimals
+	const ONE_ETH: u128 = 1_000_000_000_000_000_000;
+
+	#[test]
+	fn address_mapper_is_account_id32_mapper() {
+		assert_eq!(
+			TypeId::of::<<Runtime as pallet_revive::Config>::AddressMapper>(),
```

Comment:
Based on scope, probably easiest to import pallet_revive::Config to avoid the pallet_revive:: prefix repetition. The frame_system can be prefixed by exception if required.

Reply:
https://github.com/r0gue-io/pop-node/pull/427/commits/aad400171cdaa27cb6365b811439ad534241053e